yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -703,37 +740,61 @@ export class Sequencer {
  */
 export enum SequencerState {
   /**
-   * Will move to WAITING_FOR_TXS after a configured amount of time.
+   * Sequencer is stopped and not processing any txs from the pool.
    */
-  IDLE,
+  STOPPED = 'STOPPED',
   /**
-   * Polling the P2P module for txs to include in a block. Will move to CREATING_BLOCK if there are valid txs to include, or back to IDLE otherwise.
+   * Sequencer is awaiting the next call to work().
    */
-  WAITING_FOR_TXS,
+  IDLE = 'IDLE',
   /**
-   * Creating a new L2 block. Includes processing public function calls and running rollup circuits. Will move to PUBLISHING_CONTRACT_DATA.
+   * Synchronizing with the L2 chain.
    */
-  CREATING_BLOCK,
+  SYNCHRONIZING = 'SYNCHRONIZING',
   /**
-   * Publishing blocks to validator peers. Will move to WAITING_FOR_ATTESTATIONS.
+   * Checking if we are the proposer for the current slot.
    */
-  PUBLISHING_BLOCK_TO_PEERS,
+  PROPOSER_CHECK = 'PROPOSER_CHECK',
   /**
-   * The block has been published to peers, and we are waiting for attestations. Will move to PUBLISHING_CONTRACT_DATA.
+   * Polling the P2P module for txs to include in a block. Will move to CREATING_BLOCK if there are valid txs to include, or back to SYNCHRONIZING otherwise.
    */
-  WAITING_FOR_ATTESTATIONS,
+  WAITING_FOR_TXS = 'WAITING_FOR_TXS',
   /**
-   * Sending the tx to L1 with encrypted logs and awaiting it to be mined. Will move back to PUBLISHING_BLOCK once finished.
+   * Creating a new L2 block. Includes processing public function calls and running rollup circuits. Will move to PUBLISHING_CONTRACT_DATA.
    */
-  PUBLISHING_CONTRACT_DATA,
+  CREATING_BLOCK = 'CREATING_BLOCK',
   /**
-   * Sending the tx to L1 with the L2 block data and awaiting it to be mined. Will move to IDLE.
+   * Publishing blocks to validator peers. Will move to WAITING_FOR_ATTESTATIONS.
    */
-  PUBLISHING_BLOCK,
+  PUBLISHING_BLOCK_TO_PEERS = 'PUBLISHING_BLOCK_TO_PEERS',
   /**
-   * Sequencer is stopped and not processing any txs from the pool.
+   * The block has been published to peers, and we are waiting for attestations. Will move to PUBLISHING_CONTRACT_DATA.
    */
-  STOPPED,
+  WAITING_FOR_ATTESTATIONS = 'WAITING_FOR_ATTESTATIONS',
+  /**
+   * Sending the tx to L1 with the L2 block data and awaiting it to be mined. Will move to SYNCHRONIZING.
+   */
+  PUBLISHING_BLOCK = 'PUBLISHING_BLOCK',
+}
+
+/**
+ * The maximum number of seconds that the sequencer can be into a slot to transition to a particular state.
+ * For example, in order to transition into WAITING_FOR_ATTESTATIONS, the sequencer can be at most 3 seconds into the slot.
```

Comment:
Should the 4 here be set to the config value of max transactions per block?

Reply:
Great call. I also had it check that max time needed as a result of the config is less than the slot length.

Also I gated functionality behind and env var `SEQ_ENFORCE_TIME_TABLE`, which is now enabled by default in the helm chart values, and it passed the reorg test 🔥